File: Remove_Nth_Node_From_End_of_List/submission3.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution(object):
    def removeNthFromEnd(self, head, n):

        previous = None
        current = head
        forward = None
        temp = current

        while n!=0:
            temp = temp.next
            n -= 1
        
        while temp is not None:
            forward = current.next
            temp = temp.next
            previous = current
            current = forward
            if forward is not None:
                forward = forward.next

        if previous is None:
            forward = current.next
            current.next = None
            del current
            return forward
        else:
            previous.next = forward
            current.next = None
            del current
            return head

File: Reverse_Linked_List/submission3.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* solve(ListNode* &head, ListNode* &previous){

        if (head == NULL){
            return previous;
        } 
        ListNode* forward = head->next;
        head->next = previous;
        return solve(forward, head);
    }

    ListNode* reverseList(ListNode* head) {
        
        if (head==NULL){
            return head;
        } 
        ListNode* previous = NULL;
        return solve(head, previous);
    }
};